src/meltano/core/job/job.py

```diff
@@ -112,18 +118,18 @@ class Job(SystemModel):
     job_name: Mapped[t.Optional[str]]  # noqa: UP007
     run_id: Mapped[GUIDType]
     _state: Mapped[t.Optional[str]] = mapped_column(name="state")  # noqa: UP007
-    started_at: Mapped[t.Optional[datetime]] = mapped_column(DateTimeUTC)  # noqa: UP007
+    started_at: Mapped[datetime] = mapped_column(DateTimeUTC)
```

Comment:
The migration scripts are actually consistent with this being non-nullable, and the code elsewhere assumes the same.